crates/bevy_transform/src/components/transform.rs

```diff
@@ -185,55 +186,58 @@ impl Transform {
 
     /// Get the unit vector in the local `X` direction.
     #[inline]
-    pub fn local_x(&self) -> Vec3 {
-        self.rotation * Vec3::X
+    pub fn local_x(&self) -> Direction3d {
+        // Direction3d::new(x) panics if x is of invalid length, but quat * unit vector is length 1
+        Direction3d::new(self.rotation * Vec3::X).unwrap()
```

Comment:
These should probably use `new_unchecked` to avoid the normalization. It'd be useful to implement `Mul<Quat>` for the direction types though; with that, this would just be:
```rust
self.rotation * Direction3d::X
```
Of course, it would have to be guaranteed that rotation doesn't change the length. I think it should work, but I haven't tried with degenerate quaternions.

Reply:
This would also only work for 3D directions until we have a `Rotation2d`

Reply:
I don't know enough maths to confidently use new_unchecked (or to say that the unwrap will never be hit), so I'd appreciate if someone smarter than me could help here

Reply:
I think we should merge this with the checked version for now, and look into accelerating this (with tests) in a follow-up.